torch/distributed/tensor/_api.py

```diff
@@ -874,6 +874,13 @@ def distribute_module(
 
     torch._C._log_api_usage_once("torch.dtensor.distribute_module")
 
+    already_distributed = getattr(module, "_distribute_module_applied", False)
+    if already_distributed:
+        raise RuntimeError(
```

Comment:
I agree that we throw an error if the module is distributed twice. We better clarify this in `distribute_module` description as well.